_docs/contributing_to_a_module.md

```diff
@@ -0,0 +1,284 @@
+This will highlight and regressions that you change may have introduced.
+
+### How do I know the status of the automated tests?
+
+Within your PR if you click on the commits you will one of 3 tiny icons beside the SHA of your commit.
+
+- A green tick (Tests have passed)
+- A red cross (Tests have failed)
+- A yellow dot (Tests are still running)
+
+A PR will not be merged with failing tests, if you click on the small red cross icon it will give you additional information
+on what has actually failed. Once this is addressed you can commit your fix and continue working. If your fix is difficult and
+you need to do additional debugging check out the section in this document called **Running tests on your local machine** as this
+will allow you to run interactive debugging sessions using pry.
+
+## Running Tests on your local machine
+
+With all dependencies in place and up-to-date, run the tests:
+
+### Unit Tests
+
+```shell
+% bundle exec rake spec
+```
+
+This executes all the [rspec tests](http://rspec-puppet.com/) in the directories defined [here](https://github.com/puppetlabs/puppetlabs_spec_helper/blob/699d9fbca1d2489bff1736bb254bb7b7edb32c74/lib/puppetlabs_spec_helper/rake_tasks.rb#L17) and so on.
+rspec tests may have the same kind of dependencies as the module they are testing. Although the module defines these dependencies in its [metadata.json](./metadata.json),
+rspec tests define them in [.fixtures.yml](./fixtures.yml).
+
+### Acceptance Tests
+
+All Puppet Supported modules come with acceptance tests, which use [puppet litmus][puppet-litmus].
+Litmus supports multiple provisioners, that exist in the [provision module][provision-module].
+For information on how you can run your tests using litmus please check out the [litmus docs section][litmus-docs]
+
+## Writing Tests
+
+### Unit Tests
+
+When writing unit tests for Puppet, [rspec-puppet][] is your best friend. It provides tons of helper methods for testing your manifests against a
+catalog (e.g. contain_file, contain_package, with_params, etc). It would be ridiculous to try and top rspec-puppet's [documentation][rspec-puppet_docs]
+but here's a tiny sample:
+
+Sample manifest:
+
+```puppet
+file { "a test file":
+  ensure => present,
+  path   => "/etc/sample",
+}
+```
+
+Sample test:
+
+```ruby
+it 'does a thing' do
+  expect(subject).to contain_file("a test file").with({:path => "/etc/sample"})
+end
+```
+
+### Acceptance Tests
+
+A common pattern for acceptance tests is to create a test manifest, apply it
+twice to check for idempotency or errors, then run expectations.
+
+```ruby
+it 'does an end-to-end thing' do
+  pp =<<-EOF
+    file { "a test file":
+      ensure  => present,
+      path    => "/etc/sample",
+      content => "test string",
+    }
+  pp=<<-EOF
+
+  idempotent_apply(pp)
+end
+
+describe file("/etc/sample") do
+  it { is_expected.to contain "test string" }
+end
+```
+
+# As as Trusted Contributor
```

Comment:
This section reads a bit cart-before-the-horse as there is no other mention of review policies in here. Maybe pull out the TC-relevant bit into a link to the actual Trusted Contributor docs and only keep the general review "four eyes" principle here?